submissions/YuraZagor/Frogger/app.js

```diff
@@ -0,0 +1,108 @@
+const allEnemies = [];
+const speedLow = 200;
+const speedMedium = 275;
+const speedHigh = 350;
+const allSpeeds = [speedLow, speedMedium, speedHigh];
+const enemyRestartX = -55;
+const collisionOffset = 40
+const cellX = 101;
+const cellY = 83;
+const spriteRepositioningY = -10
+const playerStartX = 2*cellX;
+const playerStartY = 5*cellY + spriteRepositioningY ;
+const fieldWidth = 5*cellX;
+const fieldHight = 5*cellY;
+const row1 = cellY + spriteRepositioningY
+const row2 = 2*cellY + spriteRepositioningY
+const row3 = 3*cellY + spriteRepositioningY
+const rangesY = [row1, row2, row3];
+
+function Character (x, y, sprite) {
+    this.x = x;
+    this.y = y;
+    this.sprite=sprite;
+};
+
+Character.prototype.render = function() {
+    ctx.drawImage(Resources.get(this.sprite), this.x, this.y) 
+};
+
+function Enemy(x, y, sprite, speedX) {
+    Character.call(this, x, y, sprite);
+    this.speedX = speedX;
+};
+Enemy.prototype = Object.create(Character.prototype);
+
+Enemy.prototype.update = function(dt){
+    this.x += this.speedX * dt;
+    if (this.x > fieldWidth){
+        this.x = enemyRestartX;
+        this.speedX = allSpeeds[Math.floor(Math.random()*3)];
+    };
+    this.checkColision(player)
```

Comment:
`checkCollision` doesn't refer to global variable `player`, but now `update` does.
So the problem remains.
Feel free to ask specific questions to the hint from the requirement.
Yet another hint: there is a reason that you pass sprite by reference and do not refer to it directly.